test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetConsoleTestCase.cs

```diff
@@ -911,6 +911,43 @@ public async Task InstallLatestPackageInPMC(ProjectTemplate projectTemplate)
             }
         }
 
+        [NuGetWpfTheory]
+        [MemberData(nameof(GetPackagesConfigTemplates))]
+        public void VerifyInitScriptsExecution(ProjectTemplate projectTemplate)
+        {
+            EnsureVisualStudioHost();
+            using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, XunitLogger))
+            {
+                // Arrange
+                SolutionService solutionService = VisualStudio.Get<SolutionService>();
+                var nugetConsole = GetConsole(testContext.Project);
+                var projectPath = testContext.Project.FullPath;
+                var source = "https://api.nuget.org/v3/index.json";
+
+                // Act
+                nugetConsole.Execute($"install-package EntityFramework -source {source} -Verbose");
+
+                // Assert
+                Assert.True(nugetConsole.IsMessageFoundInPMC("init.ps1"), "The init.ps1 script in TestProject was not executed when the EntityFramework package was installed");
+
+                // Act
+                solutionService.CreateEmptySolution();
```

Comment:
Why do we need a new solution after entity framework has been installed?

Reply:
You can always clear the PMC if you want to make sure the check is consistent.

Reply:
Thanks for the suggestion, I've updated the code to create a new solution with clear PMC instead. Please help to review it again. Thanks.